packages/thirdweb/src/utils/any-evm/zksync/isZkSyncChain.ts

```diff
@@ -1,12 +1,27 @@
 import type { Chain } from "../../../chains/types.js";
 
-export function isZkSyncChain(chain: Chain) {
-  return (
-    chain.id === 324 ||
-    chain.id === 300 ||
-    chain.id === 302 ||
-    chain.id === 11124 ||
-    chain.id === 282 || // cronos zkevm testnet
-    chain.id === 388 // cronos zkevm mainnet
-  );
+const chainStackCache = new Map();
+
+export async function isZkSyncChain(chain: Chain) {
+  const stack = await getChainStack(chain.id);
+
+  return stack === "zksync-stack";
+}
+
+export async function getChainStack(chainId: number): Promise<string> {
+  if (chainStackCache.has(chainId)) {
+    return chainStackCache.get(chainId);
```

Comment:
we have a utility for this that handles caching much better (including caching promises)

`return withCache(async () => { ... })`

check out how we do it in `fetchPublishedContractMetadata()`

you probably want a 1 day cache on it, just in case